server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverWriteTest.java

```diff
@@ -148,17 +210,68 @@ public void testExceedingDefaultPrecisionNoRounding() {
         VectorizedRowBatch batch = (VectorizedRowBatch) batchWrapper.getData();
 
         // this value we expect to be not set and null flag turned on
         IS_NULL[0] = true;
-        assertDecimalColumnVectorCell(batch,0,14, IS_NULL, null);
+        assertDecimalColumnVectorCell(batch, 0, 14, IS_NULL, null);
+    }
+
+    @Test
+    public void testExceedingDefaultPrecisionWithNoRounding_ErrorOption() {
+        // simple test with hardcoded value assertions to make sure basic test logic itself is correct
+        columnDescriptors = getAllColumns();
+        context.setTupleDescription(columnDescriptors);
+        when(mockWriterOptions.getSchema()).thenReturn(getSchemaForAllColumns());
+        when(mockWriterOptions.getUseUTCTimestamp()).thenReturn(true);
+        context.setMetadata(mockWriterOptions);
+        configuration.set("pxf.orc.write.decimal.overflow", "error");
+        context.setConfiguration(configuration);
+
+        resolver.setRequestContext(context);
+        resolver.afterPropertiesSet();
+
+        String decimalString = "123456789012345678901234567890123456789";
+        records = new ArrayList<>(1);
+        List<OneField> record = getRecord(0, -1);
+        // reset the decimal value to a higher unsupported (>38) precision
+        record.set(14, new OneField(DataType.NUMERIC.getOID(), decimalString));
+        records.add(record);
+
+        Exception e = assertThrows(UnsupportedTypeException.class, () -> resolver.setFieldsForBatch(records));
+        assertEquals(String.format("The value %s for the NUMERIC column %s exceeds maximum precision 38.",
+                decimalString, "col14"), e.getMessage());
     }
 
+    @Test
+    public void testExceedingDefaultPrecisionWithNoRounding_RoundOption() {
+        // simple test with hardcoded value assertions to make sure basic test logic itself is correct
+        columnDescriptors = getAllColumns();
+        context.setTupleDescription(columnDescriptors);
+        when(mockWriterOptions.getSchema()).thenReturn(getSchemaForAllColumns());
+        when(mockWriterOptions.getUseUTCTimestamp()).thenReturn(true);
+        context.setMetadata(mockWriterOptions);
+        configuration.set("pxf.orc.write.decimal.overflow", "round");
+        context.setConfiguration(configuration);
+
+        resolver.setRequestContext(context);
+        resolver.afterPropertiesSet();
+
+        String decimalString = "123456789012345678901234567890123456789";
+        records = new ArrayList<>(1);
+        List<OneField> record = getRecord(0, -1);
+        // reset the decimal value to a higher unsupported (>38) precision
+        record.set(14, new OneField(DataType.NUMERIC.getOID(), decimalString));
+        records.add(record);
+
+        Exception e = assertThrows(UnsupportedTypeException.class, () -> resolver.setFieldsForBatch(records));
+        assertEquals(String.format("The value %s for the NUMERIC column %s exceeds maximum precision 38.",
```

Comment:
since it's hardcoded: 

```suggestion
        assertEquals(String.format("The value %s for the NUMERIC column col14 exceeds maximum precision 38.",
```